crates/nu-cli/src/completions/command_completions.rs

```diff
@@ -71,23 +59,24 @@ impl CommandCompletion {
                             if suggs.contains_key(&value) {
                                 continue;
                             }
-                            if matcher.matches(&name) && is_executable::is_executable(item.path()) {
+                            if is_executable::is_executable(item.path()) {
```

Comment:
Are we sure this won't noticeably affect performance if someone has a PATH with many files in it, since we're checking `is_executable` for each of them every keystroke?

I think that was pretty much the only reason I made `NuMatcher#matches`, so if it's not being used here anymore, we can probably get rid of that method entirely.

Reply:
No noticeable difference on my machine but that's a reasonable concern. I removed that because the matching filter is executed twice, we could probably  optimize both ops together.